jenkins/shared-library/vars/defaultCiPipeline.groovy

```diff
@@ -0,0 +1,163 @@
+/* groovylint-disable NestedBlockDepth */
+import java.nio.file.Paths
+
+def call(body) {
+    def params= [:]
+    body.resolveStrategy = Closure.DELEGATE_FIRST
+    body.delegate = params
+    body()
+
+    String packageRootPath = getJenkinsfilePath()
+
+    pipeline {
+        parameters {
+            gitParameter branchFilter: 'origin/(.*)',
+                    defaultValue: "${env.GIT_BRANCH}",
+                    name: 'MANUAL_GIT_BRANCH',
+                    type: 'PT_BRANCH'
+            choice name: 'PLATFORM',
+                    choices: params.platform ?: 'ubuntu',
+                    description: 'Run on specific platform'
+            booleanParam name: 'SHOULD_PUBLISH_IMAGE', description: 'true to publish image', defaultValue: false
+        }
+
+        agent {
+            label "${PLATFORM}-agent"
+        }
+
+        options {
+            ansiColor('css')
+            timestamps()
+        }
+
+        environment {
+            GITHUB_CREDENTIALS_ID = "${params.gitHubId}"
+            DOCKERHUB_CREDENTIALS_ID = 'docker-hub-token-symbolserverbot'
+            NPM_CREDENTIALS_ID = 'NPM_TOKEN_ID'
+            PYTHON_CREDENTIALS_ID = '' //TODO: need an account
+            DEV_BRANCH = 'dev'
+            RELEASE_BRANCH = 'main'
+            BUILD_SETUP_SCRIPT_FILEPATH = 'scripts/ci/setup_build.sh'
+            BUILD_SCRIPT_FILEPATH = 'scripts/ci/build.sh'
+            TEST_SETUP_SCRIPT_FILEPATH = 'scripts/ci/setup_tests.sh'
+            TEST_SCRIPT_FILEPATH = 'scripts/ci/test.sh'
+            LINTER_SCRIPT_FILEPATH = 'scripts/ci/linter.sh'
+        }
+
+        stages {
+            stage('CI pipeline') {
```

Comment:
do you envision this replacing most of catapult-client jenkins files?

Reply:
maybe some of them